### PR TITLE
Update Android Gradle plugin and cleanup

### PR DESCRIPTION
**Pull request by jshvarts**
Hey Marcin,

This PR includes:

1) Update Android Gradle plugin
2) Utilize more Kotlin stdlib functions
3) Use Android resource annotations where appropriate
4) Typos fixed

All Android tests pass. Tested the mobile app.

### CODE REVIEW

File: android/common/src/main/java/org/kotlinacademy/common/ViewExt.kt [hunk 1]
```diff
@@ -21,20 +24,20 @@ fun ImageView.loadImage(photoUrl: String) {
     Glide.with(context)
             .load(photoUrl)
             .asBitmap()
-            .transform(BorderTransformation(context, 2, context.resources.getColor(R.color.greyFrame)))
```
Comment (jshvarts) [inline android/common/src/main/java/org/kotlinacademy/common/ViewExt.kt]: resources.getColor(int colorId) is deprecated

File: android/common/src/main/java/org/kotlinacademy/common/ContextExt.kt [hunk 2]
```diff
@@ -11,18 +11,17 @@ val Context.notificationManager: NotificationManager
 fun Context.openUrl(url: String?) {
     url.nullIfBlank() ?: return
     val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
-    if (browserIntent.resolveActivity(packageManager) != null) {
```
Comment (MarcinMoskala) [inline android/common/src/main/java/org/kotlinacademy/common/ContextExt.kt]: I intentionally left it this way to keep it easier and more intuitive (especially for people from outside Kotlin world)

File: android/common/src/main/java/org/kotlinacademy/common/ContextExt.kt [hunk 3]
```diff
@@ -11,18 +11,17 @@ val Context.notificationManager: NotificationManager
...
-        startActivity(browserIntent)
-    }
+    browserIntent.resolveActivity(packageManager).let { startActivity(browserIntent) }
 }
 
 fun Context.startShareIntent(subject: String, text: String) {
-    val intent = Intent(Intent.ACTION_SEND).apply {
```
Comment (MarcinMoskala) [inline android/common/src/main/java/org/kotlinacademy/common/ContextExt.kt]: Same story - additional complexity and no real improvement.

File: android/common/src/main/java/org/kotlinacademy/common/ContextExt.kt (hunk 2 above)
Comment (jshvarts) [reply android/common/src/main/java/org/kotlinacademy/common/ContextExt.kt]: Ok--will revert this change

File: android/common/src/main/java/org/kotlinacademy/common/ContextExt.kt (hunk 3 above)
Comment (jshvarts) [reply android/common/src/main/java/org/kotlinacademy/common/ContextExt.kt]: Ok--will revert this change

**Pull request merged**